src/Symfony/Bundle/FrameworkBundle/Command/RouterDebugCommand.php

```diff
@@ -79,6 +80,12 @@ protected function execute(InputInterface $input, OutputInterface $output)
         $routes = $this->router->getRouteCollection();
 
         if ($name) {
+            $matchingRoutes = $this->findRouteNameContaining($name, $routes);
```

Comment:
I would first try to get the route by exact name before searching route names. It is useless to search the matching routes if we already have a valid name.

Reply:
done